### PR TITLE
change common debug! statements to trace!

### PR DESCRIPTION
**Pull request by ahrens**
To reduce CPU overhead, as well as log bloat, change the most common
debug! statements to trace!

Also reduce the frequency of getting the system time (Instant::new()) in
tight loops.

### CODE REVIEW

File: cmd/zfs_object_agent/zettacache/src/zettacache.rs [hunk 1]
```diff
@@ -307,14 +308,22 @@ impl MergeState {
...
+        let mut count = 0;
         while let Some(entry) = index_stream.next().await {
-            if timer.elapsed() >= *MERGE_PROGRESS_MESSAGE_INTERVAL {
+            count += 1;
+            // This can be a tight loop, so only check the elapsed time every
+            // 100 times through, so that .elapsed() doesn't take significant
+            // CPU time.
+            if count >= *MERGE_PROGRESS_CHECK_COUNT
+                && timer.elapsed() >= *MERGE_PROGRESS_MESSAGE_INTERVAL
+            {
```
Comment (mmaybee) [inline cmd/zfs_object_agent/zettacache/src/zettacache.rs:319]: It looks like you have "combined" `MERGE_PROGRESS_MESSAGE_INTERVAL` and `MERGE_PROGRESS_CHECK_COUNT`, so now we are not necessarily generating a MESSAGE every second.
1. Is once a second too frequent for a trace?
2. Should we manage these two constants separately: send a message every second, but only generate a trace! every CHECK_COUNT times?

File: cmd/zfs_object_agent/zettacache/src/zettacache.rs [hunk 2]
```diff
@@ -419,14 +428,19 @@ impl MergeState {
                 }
             }
         }
+        let mut count = 0;
         while let Some((&pc_key, &PendingChange::Insert(pc_value))) = pending_changes_iter.peek() {
-            if timer.elapsed() >= *MERGE_PROGRESS_MESSAGE_INTERVAL {
+            count += 1;
+            if count >= *MERGE_PROGRESS_CHECK_COUNT
+                && timer.elapsed() >= *MERGE_PROGRESS_MESSAGE_INTERVAL
+            {
```
Comment (mmaybee) [inline cmd/zfs_object_agent/zettacache/src/zettacache.rs:436]: Same comment as above

**Review by sdimitro — APPROVED** [review]

File: cmd/zfs_object_agent/zettacache/src/zettacache.rs (hunk 1 above)
Comment (ahrens) [reply cmd/zfs_object_agent/zettacache/src/zettacache.rs:319]: The issue isn't the `trace!` (which is a no-op by default, and even if enabled, only runs once a second), it's the `.elapsed()`, which is executed ~250,000x/sec on my system.  Assuming that we go through the loop >100x per second, we should still be sending a message once a second.

Comment (mmaybee) [reply cmd/zfs_object_agent/zettacache/src/zettacache.rs:319]: Ah, heh, I should actually read the comments! :-) 
Sorry about that. Yes, this makes sense.
Is 100 a big enough value here?  Should we consider an even larger value (e.g. 1000)?

**Review by mmaybee — APPROVED** [review]

**Pull request merged**